nmostesting/GenericTest.py

```diff
@@ -499,6 +540,103 @@ def do_test_authorization(self, api_name, test_name, error_code=401, error_type=
+
+        if not CONFIG.ENABLE_DNS_SD or CONFIG.DNS_SD_MODE != "unicast":
+            return test.DISABLED("This test cannot be performed when ENABLE_DNS_SD is False or DNS_SD_MODE is not "
+                                 "'unicast'")
+
+        if self.authorization:
+            # certificate place holder for mock auth server
+            cert = "mock_auth_cert.pem"
+            # key place holder for mock auth server
+            AUTH.key = "mock_auth_key.pem"
+
+            # create certificate and key for the mock auth server, then start the mock auth server
+            self.start_auth_server(cert, AUTH.key)
+
+            # generate mock auth server token
+            token = AUTH.generate_token([api_name])
+
+            fail = None
+            warning = None
+            headers = {"Authorization": "Bearer {}".format(token)}
+            valid, response = self.do_request("GET", url, headers=headers)
+            if not valid:
+                fail = response
+
+            if response.status_code != 401 and response.status_code != 503:
+                fail = "Incorrect response code, expected 401 or 503. Received {}".format(response.status_code)
+
+            if "WWW-Authenticate" not in response.headers:
+                fail = "Authorization error responses must include a 'WWW-Authenticate' header"
+            if not response.headers["WWW-Authenticate"].startswith("Bearer "):
+                fail = "'WWW-Authenticate' response header must begin 'Bearer'"
+
+            error_code = response.status_code
+            valid, message = self.check_error_response("GET", response, error_code)
+            if not valid:
+                fail = message
+
+            # if node responds wiuth 401, node should attempt to obtain the missing public key via the the token iss claim
+            # as specified in RFC 8414 section 3.
+            # https://specs.amwa.tv/is-10/branches/v1.0-dev/docs/4.5._Behaviour_-_Resource_Servers.html#public-keys
+            if response.status_code == 401:
+                warning = "should attempt to obtain the missing public key via the the token iss claim"
+
+                error_type = "invalid_token"
+                # Remove 'Bearer ' and tokenise
+                # https://tools.ietf.org/html/rfc6750#section-3
+                error_header_ok = False
+                auth_params = response.headers["WWW-Authenticate"][7:].split(",")
+                for param in auth_params:
+                    param_parts = param.split("=")
+                    if param_parts[0] == "error":
+                        if param_parts[1] == error_type:
+                            error_header_ok = True
+                if not error_header_ok:
+                    warning = "'WWW-Authenticate' response header should contain 'error={}'".format(error_type)
+
+            # if node responds with 503 retry GET based on the Retry-After value or maiximum up to 5 sec
```

Comment:
Perhaps the max value could be the `CONFIG.API_PROCESSING_TIMEOUT` to be flexible?

Reply:
good idea

Reply:
thanks, @prince-chrismc, changes pushed